tests/test_widget.py

```diff
@@ -355,3 +355,23 @@ def test_get_set_tooltip():
     assert widget.tooltip == "This is a tooltip."
 
 
+async def test_loading():
+    """Test setting the loading reactive."""
+
+    class LoadingApp(App):
+        def compose(self) -> ComposeResult:
+            yield Label("Hello, World")
+
+    async with LoadingApp().run_test() as pilot:
+        app = pilot.app
+        label = app.query_one(Label)
+        assert label.loading == False
+        assert len(label.query(LoadingIndicator)) == 0
+        label.loading = True
+        await pilot.pause()
+        assert len(label.query(LoadingIndicator)) == 1
+        label.loading = True  # Setting to same value is a null-op
+        label.loading = False
+        await pilot.pause()
+        assert len(label.query(LoadingIndicator)) == 0
+        label.loading = False  # Setting to same value is a null-op
```

Comment:
If setting to the same value is a no-op, we should be able to assert that the state didn't change, right?
This makes a bit more sense to me:

```suggestion
        assert len(label.query(LoadingIndicator)) == 0

        label.loading = True
        await pilot.pause()
        assert len(label.query(LoadingIndicator)) == 1

        label.loading = True  # Setting to same value is a null-op
        await pilot.pause()
        assert len(label.query(LoadingIndicator)) == 1

        label.loading = False
        await pilot.pause()
        assert len(label.query(LoadingIndicator)) == 0

        label.loading = False  # Setting to same value is a null-op
        await pilot.pause()
        assert len(label.query(LoadingIndicator)) == 0
```

Otherwise, we can just remove the no-op assignments:

```suggestion
        assert len(label.query(LoadingIndicator)) == 0
        label.loading = True
        await pilot.pause()
        assert len(label.query(LoadingIndicator)) == 1
        label.loading = False
        await pilot.pause()
        assert len(label.query(LoadingIndicator)) == 0
```

Reply:
It was more to test that setting the same value didn't crash. But it couldn't hurt to check it again.